config/initializers/okcomputer.rb

```diff
@@ -6,12 +6,37 @@
 # Required
 OkComputer::Registry.register "ruby_version", OkComputer::RubyVersionCheck.new
 OkComputer::Registry.register "redis", OkComputer::RedisCheck.new(url: ENV.fetch("REDIS_URL", Settings.redis_url))
-if Settings.rabbitmq.enabled
-  OkComputer::Registry.register "rabbit",
-    OkComputer::RabbitmqCheck.new(hostname: Settings.rabbitmq.hostname,
+
+class RabbitQueueExistsCheck < OkComputer::Check
+  attr_accessor :queue_names, :conn
+
+  def initialize(queue_names)
+    self.queue_names = Array(queue_names)
+    self.conn = Bunny.new(hostname: Settings.rabbitmq.hostname,
       vhost: Settings.rabbitmq.vhost,
       username: Settings.rabbitmq.username,
       password: Settings.rabbitmq.password)
+  end
+
+  def check
+    conn.start
+    status = conn.status
+    missing_queue_names = queue_names.reject { |queue_name| conn.queue_exists?(queue_name) }
+    if missing_queue_names.empty?
+      mark_message "'#{queue_names.join(", ")}' exists, connection status: #{status}"
+    else
+      mark_message "'#{missing_queue_names.join(", ")}' does not exist"
+      mark_failure
+    end
+    conn.close
```

Comment:
Should `conn.close` be within a method-level `ensure` block to make sure the connection is always closed?

Reply:
Not sure. It isn't in https://github.com/sportngin/okcomputer/blob/master/lib/ok_computer/built_in_checks/rabbitmq_check.rb